collections/examples/multi_index.go

```diff
@@ -0,0 +1,60 @@
+package examples
+
+import (
+	"context"
+	"cosmossdk.io/collections"
+	"cosmossdk.io/collections/indexes"
+)
+
+var validatorCodec collections.ValueCodec[Validator] = nil // faux collections.ValueCodec, this is most likely a proto so you'll be using codec.ProtoValueCodec
+
+type Validator struct {
+	Power   uint64
+	Moniker string
+}
+
+type ValidatorsIndexes struct {
+	// Power is the index of validators by their power.
+	Power *indexes.Multi[uint64, ValAddress, Validator]
+}
+
+// IndexesList implements collections.Indexes
+func (i ValidatorsIndexes) IndexesList() []collections.Index[ValAddress, Validator] {
+	return []collections.Index[ValAddress, Validator]{i.Power}
+}
+
+type Keeper2 struct {
+	Validators *collections.IndexedMap[ValAddress, Validator, ValidatorsIndexes]
+}
+
+func NewKeeper2() Keeper2 {
+	sb := collections.NewSchemaBuilder(nil)
+	return Keeper2{
+		Validators: collections.NewIndexedMap(
```

Comment:
this creation is a bit intense. seems a bit complicated to me. It is write it once reuse it everywhere but worried some users may run into issues. Thoughts?

Reply:
I'm not sure if it can be simplified, for instances this is cosmwasm indexed map instantiation:
<img width="955" alt="image" src="https://user-images.githubusercontent.com/98415576/214178005-e0d98a43-1b02-48ef-88bd-e054e7a542f8.png">

(sorry couldn't find some code examples)

Basically the only thing extra we do here is passing key encoders as parameters because go's generics aren't as powerful as rust's.

The only way to simplify this would be to instantiate key encoders etc through reflection.... maybe doable but probably in another PR...

Reply:
true forgot how it was done in cosmwasm.

Reply:
Just because this is how CW does it, doesn't mean it provides a better UX. I'm sure there are ways we can think of to improve it, no?

Reply:
yeah... maybe there are ways